pt_os_web_portal/backend/routes.py

```diff
@@ -557,3 +563,23 @@ def post_disable_ap_mode():
 def get_client_should_switch_network():
     logger.debug("Route '/should-switch-networks'")
     return jdumps(should_switch_network(request))
+
+
+@app.route("/start-vnc-wpa-gui", methods=["POST"])
+def post_start_vnc_wpa_gui():
+    logger.debug("Route '/start-vnc-wpa-gui'")
+    service_restart(SystemService.VncWpaGui)
+    return "OK"
```

Comment:
would be good if this could return an error when the service fails to start

Reply:
not sure about this... it would mean we'd need to wait a few seconds in order to determine the service status; the service goes through a `running` and `exited` states, where `running` means it's running the `ExecStart` command from the unit file and `exited` means it finished running those with no errors... I'd assume we'd want to wait until the state is either `failed` (or something like that) or `exited`. We could also have the frontend poll to learn about the service state?

Reply:
yeah, fair enough, I think vnc_wpa_gui_url should maybe error out then when it can't get the status? that might not be ideal either, but the current state seems to be that this whole thing can be totally broken and the frontend will just say it's loading for ever. it would be good if the backend can detect and respond with an error, but as a last resort we could add a timeout on the frontend if the url is not forthcoming